### ISSUE TITLE
aarch64 assembler error

### ISSUE THREAD

**Issue by vielmetti** [facebook/hhvm#7910]
### HHVM Version

master at ced4801

### Standalone code, or other way to reproduce the problem

Follow build instructions with "make -j 96" on Ubuntu 16.04 LTS / Packet Type 2A (Cavium ThunderX) aarch64/arm64 system

### Expected result

Build competes successfully.

### Actual result

Build fails with this message

```
/tmp/camlasm46a45a.s: Assembler messages:
/tmp/camlasm46a45a.s:38621: Error: immediate out of range
File "parser/full_fidelity_validated_syntax.ml", line 1:
Error: Assembler error, input left in file /tmp/camlasm46a45a.s
Command exited with code 2.
```

Also reported on the "HHVM on AARCH64" mailing list at 

https://groups.google.com/forum/#!topic/hhvm-arm/O6musungq1c

which points to commits 50d383850ecb483efd0ef21a32e07907a743807f and 191828d09db501ed75c3fd19243536f7f2d46410 which when reverted allow a clean build on aarch64.

**Comment by apinski-cavium** [comment]
.L4580: sub     x27, x27, #8728

(gdb) p/x 8728
$1 = 0x2218

**Comment by swalk-cavium** [comment]
This is an issue in the ocaml third-party submodule.  It is generating invalid immediate constants for ARMv8 as @apinski-cavium notes.

**Comment by vielmetti** [comment]
This is addressed in #7918 , as far as I can tell.

**Comment by vielmetti** [comment]
After #7918 was applied, there is a successful build of this; closing.
